### PR TITLE
add options for sensitivity calculations in basic statistics pp

### LINKED ISSUE #776: options for sensitivity calculations in basic statistics pp
--------
Issue Description
--------
This is a new feature to allow the users to input additional node ```<multipleLinear>``` to compute the sensitivity and variance dependent sensitivity calculations. The default is True for ```<multipleLinear>``` which is consistent with the current devel branch. When ```<multipleLinear>```  is False, the direct one target to one feature is used to compute the sensitivities. 

##### Please attach the input file(s) that generate this error.  The simpler the input, the faster we can find the issue.

----------------
For Change Control Board: Issue Review
----------------
This review should occur before any development is performed as a response to this issue.
- [x] 1. Is it tagged with a type: defect or improvement?
- [x] 2. Is it tagged with a priority: critical, normal or minor?
- [x] 3. If it will impact requirements or requirements tests, is it tagged with requirements?
- [x] 4. If it is a defect, can it cause wrong results for users? If so an email needs to be sent to the users.
- [x] 5. Is a rationale provided? (Such as explaining why the improvement is needed or why current code is wrong.)

-------
For Change Control Board: Issue Closure
-------
This review should occur when the issue is imminently going to be closed.
- [x] 1. If the issue is a defect, is the defect fixed?
- [x] 2. If the issue is a defect, is the defect tested for in the regression test system?  (If not explain why not.)
- [x] 3. If the issue can impact users, has an email to the users group been written (the email should specify if the defect impacts stable or master)?
- [x] 4. If the issue is a defect, does it impact the latest stable branch? If yes, is there any issue tagged with stable (create if needed)?
- [x] 5. If the issue is being closed without a merge request, has an explanation of why it is being closed been provided?

### PR DESCRIPTION
**Pull request by wangcj05**
--------
Pull Request Description
--------
Add new feature
##### What issue does this change request address? (Use "#" before the issue to link it, i.e., #42.)
Closes #776

##### What are the significant changes in functionality due to this change request?

----------------
For Change Control Board: Change Request Review
----------------
The following review must be completed by an authorized member of the Change Control Board.
- [x] 1. Review all computer code.
- [x] 2. If any changes occur to the input syntax, there must be an accompanying change to the user manual and xsd schema. If the input syntax change deprecates existing input files, a conversion script needs to be added (see Conversion Scripts).
- [x] 3. Make sure the Python code and commenting standards are respected (camelBack, etc.) - See on the [wiki](https://github.com/idaholab/raven/wiki/RAVEN-Code-Standards#python) for details.
- [x] 4. Automated Tests should pass, including run_tests, pylint, manual building and xsd tests. If there are changes to Simulation.py or JobHandler.py the qsub tests must pass.
- [x] 5. If significant functionality is added, there must  be tests added to check this. Tests should cover all possible options.  Multiple short tests are preferred over one large test. If new development on the internal JobHandler parallel system is performed, a cluster test must be added setting, in <RunInfo> XML block, the node ```<internalParallel>``` to True.
- [x] 6. If the change modifies or adds a requirement or a requirement based test case, the Change Control Board's Chair or designee also needs to approve the change.  The requirements and the requirements test shall be in sync.
- [x] 7. The merge request must reference an issue.  If the issue is closed, the issue close checklist shall be done.
- [x] 8. If an analytic test is changed/added is the the analytic documentation updated/added?

### CODE REVIEW

File: framework/PostProcessors/BasicStatistics.py [hunk 1]
```diff
@@ -105,6 +105,9 @@ class cls.
     biasedInput = InputData.parameterInputFactory("biased", contentType=InputData.StringType)
     inputSpecification.addSub(biasedInput)
 
+    multipleLinearInput = InputData.parameterInputFactory("multipleLinear", contentType=InputData.StringType)
```
Comment (alfoa) [inline framework/PostProcessors/BasicStatistics.py]: The type should be boolean type and not string.

File: framework/PostProcessors/BasicStatistics.py [hunk 2]
```diff
@@ -314,6 +318,9 @@ def _handleInput(self, paramInput):
       elif tag == "dataset":
         if child.value.lower() in utils.stringsThatMeanTrue():
           self.outputDataset = True
+      elif tag == "multipleLinear":
+        if child.value.lower() in utils.stringsThatMeanFalse():
```
Comment (alfoa) [inline framework/PostProcessors/BasicStatistics.py]: If you follow what I wrote just above, this if statement is not needed

File: doc/user_manual/postprocessor.tex [hunk 3]
```diff
@@ -166,6 +166,11 @@ \subsubsection{BasicStatistics}
     \end{itemize}
   \default{False}
   %
+  \item \xmlNode{multipleLinear}, \xmlDesc(boolean, optional field), this node can be used when the users do not want
+    to compute the multiple linear sensitivities. This method can be used when the input features depend on each
```
Comment (alfoa) [inline doc/user_manual/postprocessor.tex]: This documentation should be extended and improved. For example,  "the users do not want to compute the multiple linear sensitivities"... what do we use instead? It is not explained and the user does not know.

Comment (alfoa) [reply doc/user_manual/postprocessor.tex]: Also the name of the node is not self-explanatory

File: framework/PostProcessors/BasicStatistics.py [hunk 4]
```diff
@@ -967,22 +974,29 @@ def sensitivityCalculation(self,featVars, targVars, featSamples, targSamples, in
       @ In, intersectionSet, boolean, True if some target variables are in the list of features
       @ Out, da, xarray.DataArray, contains the calculations of sensitivity coefficients
     """
-    if not intersectionSet:
-      senMatrix = LinearRegression().fit(featSamples,targSamples).coef_
+    if self.multipleLinear:
+      if not intersectionSet:
+        senMatrix = LinearRegression().fit(featSamples,targSamples).coef_
```
Comment (alfoa) [inline framework/PostProcessors/BasicStatistics.py]: comments/...add explain what it is done...

File: framework/PostProcessors/BasicStatistics.py (hunk 1 above)
Comment (wangcj05) [reply framework/PostProcessors/BasicStatistics.py]: fixed

File: framework/PostProcessors/BasicStatistics.py (hunk 2 above)
Comment (wangcj05) [reply framework/PostProcessors/BasicStatistics.py]: fixed

File: doc/user_manual/postprocessor.tex (hunk 3 above)
Comment (wangcj05) [reply doc/user_manual/postprocessor.tex]: updated, and I changed the name to ```multipleFeatures```

File: framework/PostProcessors/BasicStatistics.py (hunk 4 above)
Comment (wangcj05) [reply framework/PostProcessors/BasicStatistics.py]: added.

**Comment by wangcj05** [comment]
@alfoa I have addressed your comments. If you have time, please review these changes.

**Comment by alfoa** [comment]
Checklist passed... 

Merging...

**Pull request merged**